packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexContainer.tsx

```diff
@@ -108,6 +124,63 @@ export const RecordIndexContainer = ({
     [columnDefinitions, setTableColumns],
   );
 
+  const navigate = useNavigate();
+
+  const { records: views } = usePrefetchedData<View>(PrefetchKey.AllViews);
+
+  const currentViewId = useRecoilValue(
+    currentViewIdComponentState({
+      scopeId: recordIndexId,
+    }),
+  );
+
+  const view = findView({
```

Comment:
can't we use useGetCurrentView here? also I'm worried about re-renderings here, shoudn't be this either be in an effect, computing a state, either using recoilCallback?